kong/plugins/key-auth/handler.lua

```diff
@@ -179,9 +237,43 @@ local function do_authentication(conf)
     return nil, { status = 500, message = "An unexpected error occurred" }
   end
 
-  set_consumer(consumer, credential)
+  if conf.validate_signature then
+    -- 4. Verify sha256
```

Comment:
this comment seems a copy-paste left over.

Also this entire check should be moved up, before the `Success, this request is authenticated` comment.